shared/modules/telemetry_module.go

```diff
@@ -0,0 +1,54 @@
+package modules
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type (
+	TelemetryModule interface {
+		Module
+
+		// Time series
+		GetTimeSeriesAgent() TimeSeriesAgent
+		GetEventMetricsAgent() EventMetricsAgent
+	}
+
+	// Interface for time series agent (prometheus)
+	TimeSeriesAgent interface {
+		// Counters
+
+		// Register a counter by name
+		RegisterCounter(name string, description string)
+		// Increment the counter
+		IncCounter(name string)
+
+		// Gauges
+
+		// Register a gauge by name
+		RegisterGauge(name string, description string)
+		// Set sets the Gauge to an arbitrary value.
+		SetGauge(name string, value float64) prometheus.Gauge

```

Comment:
This is somewhat of a historic thing that remained. I initially was interesting in  chaining telemetry calls to make use of `WithLabelValues` and `WithLabels` calls. So we could do away with this change as it's not used anywhere at the moment, but we can make use of it if we keepit, and it would help us do something like the following:
```
m.SetGauge("x", 0).WithLabelValues(..).Add(10)
```